apps/grader-host/src/lib/receiveFromQueue.js

```diff
@@ -1,22 +1,39 @@
 // @ts-check
 import { setTimeout as sleep } from 'node:timers/promises';
-import * as path from 'path';
 
 import {
   ReceiveMessageCommand,
   ChangeMessageVisibilityCommand,
   DeleteMessageCommand,
 } from '@aws-sdk/client-sqs';
-import { Ajv } from 'ajv';
 import * as async from 'async';
-import fs from 'fs-extra';
+import { z } from 'zod';
 
 import * as Sentry from '@prairielearn/sentry';
 
 import { config } from './config.js';
 import globalLogger from './logger.js';
 
-let messageSchema = null;
+const GradingJobMessageSchema = z.object({
+  /** The unique ID for this job. */
+  jobId: z.string(),
+  /** The Docker image that the grading job will be executed in. */
+  image: z.string(),
+  /** The entrypoint for the container. */
+  entrypoint: z.string(),
+  /** The number of seconds after which the grading job will time out. */
+  timeout: z.number(),
+  /** Whether or not the container should have internet access. */
+  enableNetworking: z.boolean(),
+  /** Environment variables for the container. */
+  environment: z.record(z.string()),
+  /** The AWS S3 bucket containing this job's files. */
+  s3Bucket: z.string(),
+  /** The root key for the job's files. */
+  s3RootKey: z.string(),
+});
```

Comment:
I would feel more comfortable with these changes if proper types are defined in here (at least in JSDoc). 
https://github.com/PrairieLearn/PrairieLearn/blob/3ab74239957a3609d0cd30aab07f4d4410a9e9ee/apps/prairielearn/src/lib/externalGraderSqs.js#L97-L109

I won't block on this, though.

Reply:
We don't yet have a great way to share types between apps. We could of course create a new package... but I don't think that's strictly necessary here.

Reply:
I don't necessarily mean sharing the types, only declaring them there for easier comparison.